tests/cp.go

```diff
@@ -0,0 +1,147 @@
+					command.Run(o, "exec", testContainerName, "ln", "-s", containerFilepath, containerSymlink)
+					fileDir := ffs.CreateTempDir("finch-test")
+					path := filepath.Join(fileDir, filename)
+					ginkgo.DeferCleanup(os.RemoveAll, fileDir)
+
+					command.Run(o, "cp", link, fmt.Sprintf("%s:%s", testContainerName, containerSymlink), path)
+					fileShouldExist(path, content)
+				})
+			}
+
+			ginkgo.It("should not be able to copy nonexistent file from host to container", func() {
+				fileDir := ffs.CreateTempDir("finch-test")
+				ginkgo.DeferCleanup(os.RemoveAll, fileDir)
+
+				cmdOut := command.RunWithoutSuccessfulExit(o, "cp", filepath.Join(fileDir, filename), containerResource)
+				gomega.Expect(cmdOut.Err.Contents()).To(gomega.ContainSubstring("no such file or directory"))
+				fileShouldNotExistInContainer(o, testContainerName, containerFilepath)
+			})
+
+			ginkgo.It("should not be able to copy nonexistent file from container to host", func() {
+				fileDir := ffs.CreateTempDir("finch-test")
+				path := filepath.Join(fileDir, filename)
+				ginkgo.DeferCleanup(os.RemoveAll, fileDir)
+
+				cmdOut := command.RunWithoutSuccessfulExit(o, "cp", containerResource, path)
+				gomega.Expect(cmdOut.Err.Contents()).To(gomega.ContainSubstring("no such file or directory"))
+				fileShouldNotExist(path)
+			})
+		})
+
+		ginkgo.Context("when the container is not running", func() {
+			ginkgo.It("should not be able to copy file from host to container", func() {
+				command.Run(o, "run", "--name", testContainerName, defaultImage)
+				path := ffs.CreateTempFile(filename, content)
+				ginkgo.DeferCleanup(os.RemoveAll, filepath.Dir(path))
+				cmdOut := command.RunWithoutSuccessfulExit(o, "cp", path, containerResource)
+				gomega.Expect(cmdOut.Err.Contents()).To(gomega.ContainSubstring("expected container status running"))
+			})
+
+			ginkgo.It("should not be able to copy file from container to host", func() {
+				cmd := fmt.Sprintf("echo -n %s > %s", content, containerFilepath)
+				command.Run(o, "run", "--name", testContainerName, defaultImage, "sh", "-c", cmd)
+				fileDir := ffs.CreateTempDir("finch-test")
+				path := filepath.Join(fileDir, filename)
+				ginkgo.DeferCleanup(os.RemoveAll, fileDir)
+				cmdOut := command.RunWithoutSuccessfulExit(o, "cp", containerResource, path)
+				gomega.Expect(cmdOut.Err.Contents()).To(gomega.ContainSubstring("expected container status running"))
+			})
+		})
+	})
+}
+
+func fileShouldExist(path string, content string) {
+	gomega.Expect(path).To(gomega.BeARegularFile())
+	actualContent, err := os.ReadFile(filepath.Clean(path))
+	gomega.Expect(err).ToNot(gomega.HaveOccurred())
+	gomega.Expect(string(actualContent)).To(gomega.Equal(content))
+}
+
+func fileShouldNotExist(path string) {
+	gomega.Expect(path).ToNot(gomega.BeAnExistingFile())
+}
+
+func fileShouldExistInContainer(o *option.Option, containerName string, path string, content string) {
```

Comment:
nit:
```suggestion
func fileShouldExistInContainer(o *option.Option, containerName, path, content string) {
```
Same as other functions.